autoscheduler/frontend/src/tests/ui/CourseSelectColumn.test.tsx

```diff
@@ -19,16 +23,20 @@ function ignoreInvisible(content: string, element: HTMLElement, query: string |
 
 describe('CourseSelectColumn', () => {
   describe('Adds a course card', () => {
-    test('wehn the Add Course button is clicked', () => {
+    test('when the Add Course button is clicked', async () => {
       // arrange
       const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
       store.dispatch(setTerm('201931'));
 
+      // sessions/get_saved_courses
+      fetchMock.mockResponseOnce(JSON.stringify([]));
+
       const { getByText, getAllByText } = render(
         <Provider store={store}>
           <CourseSelectColumn />
         </Provider>,
       );
+      await new Promise(setImmediate);
```

Comment:
What does this do? And why do we need it

Reply:
This waits for all pending promises to be resolved, since we have to wait for all of the actions called by `replaceCourseCards` to finish if this isn't here then the lines after the dispatch will run before it's completed

Reply:
Actually I don't think it's needed, I've been testing without it and it seems to work fine. Since the action is synchronous it should have to wait until after it finishes to continue